presto-main/src/main/java/io/prestosql/server/security/BasicAuthCredentials.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.security;
+
+import com.google.common.base.Splitter;
+
+import javax.servlet.http.HttpServletRequest;
+
+import java.util.Base64;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.emptyToNull;
+import static com.google.common.base.Strings.nullToEmpty;
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+import static java.nio.charset.StandardCharsets.ISO_8859_1;
+import static java.util.Objects.requireNonNull;
+
+public class BasicAuthCredentials
+{
+    private final String user;
+    private final Optional<String> password;
+
+    public static Optional<BasicAuthCredentials> extractBasicAuthCredentials(HttpServletRequest request)
+            throws AuthenticationException
+    {
+        // This handles HTTP basic auth per RFC 7617. The header contains the
+        // case-insensitive "Basic" scheme followed by a Base64 encoded "user:pass".
+        String header = nullToEmpty(request.getHeader(AUTHORIZATION));
```

Comment:
From an API perspective, it's cleaner to pass in the header value, as that would make testing or other reuse easier (single string vs full request object).